akka-actor/src/main/scala/akka/actor/RepointableActorRef.scala

```diff
@@ -236,26 +245,10 @@ private[akka] class UnstartedCell(val systemImpl: ActorSystemImpl,
     lock.lock // we cannot lose system messages, ever, and we cannot throw an Error from here as well
     try {
       val cell = self.underlying
-      if (cellIsReady(cell)) {
+      if (cellIsReady(cell))
         cell.sendSystemMessage(msg)
-      } else {
-        // systemMessages that are sent during replace need to jump to just after the last system message in the queue, so it's processed before other messages
-        val wasEnqueued = if ((self.lookup ne this) && (self.underlying eq this) && !queue.isEmpty()) {
```

Comment:
This is something that I'm still unsure about. What was the purpose of the `(self.lookup ne this) && (self.underlying eq this)` check? I tried to add it back like:

    if ((self.lookup ne this) && (self.underlying eq this)) {
          sysmsgQueue ::= msg

but that cause `akka.routing.RoutingSpec` to fail due to dropped Watch sys msg.

In the old code, `queue.offer` will never return `false` so `wasEnqueued` will always be `true`.